trunk/SUAVE/Methods/Aerodynamics/Supersonic_Zero/Drag/compressibility_drag_total.py

```diff
@@ -215,18 +215,40 @@ def drag_div(Mc_ii,wing,k,cl,Sref_main):
 
         # Divergence mach number
         MDiv = np.array([[0.95]] * len(Mc_ii))
-        mcc = np.array([[0.93]] * len(Mc_ii))
+        mcc  = np.array([[0.93]] * len(Mc_ii))
 
     else:
         # Unpack wing
         t_c_w   = wing.thickness_to_chord
-        sweep_w = wing.sweeps.quarter_chord
+        if len(wing.Segments.keys())>0: # if wing has segments
+            symm           = wing.symmetric
+            semispan       = wing.spans.projected*0.5 * (2 - symm)
+            root_chord     = wing.chords.root
+            num_segments   = len(wing.Segments.keys())     
+            
+            weighted_sweep = 0
+            Sref           = 0
+            for i_segs in xrange(num_segments): 
+                if i_segs == num_segments-1:
```

Comment:
you know, you could make the loop one less...

Reply:
see comment below. Issue resolved